### PR TITLE
gg-33259: calculate total table row count from overridden values.

### PR DESCRIPTION
**Pull request by Berkof**

### CODE REVIEW

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsHelper.java [hunk 1]
```diff
@@ -149,6 +149,18 @@ public static ObjectStatisticsImpl aggregateLocalStatistics(
             colStats.put(col.getName(), stat);
         }
 
+        long overridedRowCnt = -1;
+
+        for (StatisticsColumnConfiguration ccfg : cfg.columns().values()) {
+            if (ccfg.overrides() != null && ccfg.overrides().total() != null) {
+                Long colRowCnt = ccfg.overrides().total();
+
+                overridedRowCnt = (overridedRowCnt > colRowCnt) ? overridedRowCnt : colRowCnt;
```
Comment (ygerzhedovich) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsHelper.java:158]: Why not Math.max(overridedRowCnt, cloRowCnt)?

Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsHelper.java:158]: To allow user override this value to lower it.

Comment (ygerzhedovich) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsHelper.java:158]: I didn't catch your comment. I mean the construction could be amended to Math.max.

Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsHelper.java:158]: Ok, if I did it - how user can lower rowCount to make it less than actual rows count? Or you trying to say that we'll never need it?

**Pull request merged**